mitmproxy/net/server_spec.py

```diff
@@ -3,12 +3,11 @@
 """
 import re
 from functools import cache
-from typing import Literal
 
 from mitmproxy.net import check
 
 ServerSpec = tuple[
-    Literal["http", "https", "tcp", "tls", "dns"],
+    str,
```

Comment:
We're already checking the returned scheme in `mode_specs.py`, so instead of adding `h1`, `h2` and `h3` (which will be checked again in `QuicMode` anyways), I think it's better to make `server_spec.parse` scheme-agnostic.